File: CPP/inheritence/multiple.cpp
#include <iostream>
using namespace std;

class student
{
	int roll;
	string name;
	public:
	student()
	{
		cout<<"student constructor"<<endl;
		cout<<"enter details roll and name"<<endl;
		cin>>roll;
		cin>>name;
	}
	void print()
	{
		cout<<roll<<endl<<name<<endl;
	}
	~student()
	{
		cout<<"student destructor"<<endl;
	}
};
class Branch
{
	string branch;
	public:
	Branch()
	{
		cout<<"Branch constructor"<<endl;
		cout<<"enter branch"<<endl;
		cin>>branch;
	}
	void print()
	{
		cout<<branch<<endl;
	}
	~Branch()
	{
		cout<<"Branch destructor"<<endl;
	}
};
class sport:public student , public Branch
{
	string sportname;
	public:
	sport()
	{
                cout<<"sport constructor"<<endl;
                cout<<"enter sport"<<endl;
                cin>>sportname;
        }
        void print()
        {
		student::print();
		Branch::print();
                cout<<sportname<<endl;
        }
        ~sport()
        {
                cout<<"sport destructor"<<endl;
        }

};	
int main()
{
	sport obj;
	obj.print();
}

File: PYTHON/inheritence/multiple.py
class student():
    def name(self,name):
        self.name=name
    def stu_print(self):
        print(self.name)
class Branch():
    def branch(self,branch):
        self.branch=branch
    def bran_print(self):
        print(self.branch)

class sport(student,Branch):
    def sport(self,sport):
        self.sport=sport
    def print(self):
        super().stu_print()
        super().bran_print()
        print(self.sport)

obj=sport()
obj.name("bhaskar")
obj.branch("ECE")
obj.sport("cricket")
#obj.stu_print()
obj.print()
